packages/repmgr13/repmgr.spec

```diff
@@ -64,21 +78,23 @@ command line option handling.
 
 %build
 
-PG_CONFIG=/usr/bin/pg_config ./configure
+PG_CONFIG=%{pginstdir}/bin/pg_config ./configure
```

Comment:
Is there an advantage to changing this?

Reply:
Good question.  I did this to try to minimize my changes on top of the upstream source rpm.

In this commit:

53bcb938f5c9499f8865c1af908b6a120ab2ac94 
```
    Commit the contents of repmgr13-5.2.1-1.el8.src.rpm

    We've excluded the repmgr-5.2.1.tar.bz2 and renamed tmpm5LtCB.spec to
    repmgr.spec.

    It was downloaded from:
    https://dl.enterprisedb.com/default/release/browse/rpm/src_packages/rhel/8Server/x86_64/13/repmgr13-5.2.1-1.el8.src.rpm
```

They made this change on top of what they had in repmgr10-4.0.6:

```diff
diff --git a/packages/repmgr13/repmgr.spec b/packages/repmgr13/repmgr.spec
index 35360aa..1fa02cf 100644
--- a/packages/repmgr13/repmgr.spec
+++ b/packages/repmgr13/repmgr.spec
@@ -1,5 +1,6 @@
-%global pgmajorversion 10
+%global pgmajorversion 13
 %global pgpackageversion %(echo %{pgmajorversion} | tr -d .)
+%global pginstdir /usr/pgsql-%{pgmajorversion}
...
@@ -64,12 +75,12 @@ command line option handling.

 %build

-PG_CONFIG=/usr/bin/pg_config ./configure
+PG_CONFIG=%{pginstdir}/bin/pg_config ./configure

-%{__make} PG_CONFIG=/usr/bin/pg_config USE_PGXS=1 %{?_smp_mflags}
+%{__make} PG_CONFIG=%{pginstdir}/bin/pg_config USE_PGXS=1 %{?_smp_mflags}

 %install
-%{__mkdir} -p %{buildroot}/usr/bin/
+%{__mkdir} -p %{buildroot}/%{pginstdir}/bin/
 %if %{systemd_enabled}
 # Use new %%make_install macro:
 USE_PGXS=1 %make_install  DESTDIR=%{buildroot}
...
```

I could remove these changes that added the `pginstdir` all over the place since they're assuming you need to install a postgresql version that installs to a versioned path or just hack it like I did.  What do you think?

Reply:
Being closer to the upstream source is probably better

Reply:
> Being closer to the upstream source is probably better

yes, it's unfortunate they changed their upstream src rpm to specify a specific version path for the postgresql binaries (such as /usr/pgsql-13) and related things as that conflicts with pg 13 being the default pg on ubi9 / centos9 but I think my change tries to minimize my changes on top of theirs.  

🤷